nexus/reconfigurator/planning/src/mgs_updates/mod.rs

```diff
@@ -511,6 +666,218 @@ fn try_make_update_sp(
     })
 }
 
+/// Determine if the given baseboard needs an SP update and, if so, returns it.
+fn try_make_update_rot(
+    log: &slog::Logger,
+    baseboard_id: &Arc<BaseboardId>,
+    inventory: &Collection,
+    current_artifacts: &TufRepoDescription,
+) -> Option<PendingMgsUpdate> {
+    let Some(sp_info) = inventory.sps.get(baseboard_id) else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing SP info from inventory)";
+            baseboard_id
+        );
+        return None;
+    };
+
+    let Some(rot_state) = inventory.rots.get(baseboard_id) else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing RoT state from inventory)";
+            baseboard_id
+        );
+        return None;
+    };
+
+    let active_slot = rot_state.active_slot;
+
+    let active_caboose = match active_slot {
+        RotSlot::A => CabooseWhich::RotSlotA,
+        RotSlot::B => CabooseWhich::RotSlotB,
+    };
+
+    let Some(active_caboose) =
+        inventory.caboose_for(active_caboose, baseboard_id)
+    else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (missing active slot {active_slot} caboose from inventory)";
+            baseboard_id,
+        );
+        return None;
+    };
+
+    let Ok(expected_active_version) = active_caboose.caboose.version.parse()
+    else {
+        warn!(
+            log,
+            "cannot configure RoT update for board \
+             (cannot parse current active version as an ArtifactVersion)";
+            baseboard_id,
+            "found_version" => &active_caboose.caboose.version,
+        );
+        return None;
+    };
+
+    let board = &active_caboose.caboose.board;
+    let Some(rkth) = &active_caboose.caboose.sign else {
```

Comment:
We spoke about comparing against the CMPA page's rkth field instead. But unfortunately, all I get from inventory is a base64 encoded string. I did not find a way in omicron or hubtools to decode that safely into a known structure. While a worthy goal to compare the artifact's sign to the CMPA's rkth field, in the interest of time, I think for now I'd like to compare against the inventory caboose's sign field. 

I can write up a follow up issue to tackle this at a later time.